dex-auth/dex-crds/overlays/github/config-map.yaml

```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: dex
+data:
+  config.yaml: |
+    issuer: $(issuer)
+    storage:
+      type: kubernetes
+      config:
+        inCluster: true
+    web:
+      http: 0.0.0.0:5556
+    logger:
+      level: "debug"
+      format: text
+    connectors:
+      - type: github
+        # Required field for connector id.
+        id: github
+        # Required field for connector name.
+        name: GitHub
+        config:
+          # Credentials can be string literals or pulled from the environment.
+          clientID: $(github_client_id)
+          clientSecret: $(github_client_secret)
+          redirectURI: http://dex.auth.svc.cluster.local:5556/dex/callback
+          # Optional organizations and teams, communicated through the "groups" scope.
+          #
+          # NOTE: This is an EXPERIMENTAL config option and will likely change.
+          #
+          orgs:
+          - name: $(github_org_name)
+          # Required ONLY for GitHub Enterprise.
```

Comment:
Any chance you have tested with normal external github instead of enterprise?

Reply:
It should work as long as keeping parameter `github_hostname` empty or don't specify it. since dex will change GitHub api endpoint only when `github_hostname` is not empty:

https://github.com/dexidp/dex/blob/d87cf1c92474263ade8b5909204024fb0d23f621/connector/github/github.go#L89-L97

Will give it try now

Reply:
it works well against github.com when NOT specifying `github_hostname`:

```YAML
    - kustomizeConfig:
      overlays:
      - github
      parameters:
      - name: namespace
        value: auth
      - name: issuer
        value: http://dex.auth.svc.cluster.local:5556/dex
      - name: client_id
        value: kubeflow-oidc-authservice
      - name: oidc_redirect_uris
        value: '["/login/oidc"]'
      - name: github_client_id
        value: <configured GitHub OAuth app client id>
      - name: github_client_secret
        value: <configured GitHub OAuth app client secret>
      - name: github_org_name
        value: <github organization name>
      - name: github_use_login_as_id
        value: true
      repoRef:
        name: manifests
        path: dex-auth/dex-crds
```

I will update the comment in the ConfigMap of this overlay to clarify it